skyportal/handlers/api/source.py

```diff
@@ -77,1919 +77,2067 @@
+                            Source.group_id == group["id"],
+                        )
+                        .first()
+                    )
+                    if source_table_row is not None:
+                        group["active"] = source_table_row.active
+                        group["requested"] = source_table_row.requested
+                        group["saved_at"] = source_table_row.saved_at
+                        group["saved_by"] = (
+                            source_table_row.saved_by.to_dict()
+                            if source_table_row.saved_by is not None
+                            else None
+                        )
+
+            if include_color_mag:
+                obj_list[-1]["color_magnitude"] = get_color_mag(
+                    obj_list[-1]["annotations"]
+                )
+        query_results["sources"] = obj_list
+
+    query_results = recursive_to_dict(query_results)
+    if includeGeoJSON:
+        # features are JSON representations that the d3 stuff understands.
+        # We use these to render the contours of the sky localization and
+        # locations of the transients.
+
+        features = []
+
+        # useful for testing visualization
+        # import numpy as np
+        # for xx in np.arange(30, 180, 30):
+        #   features.append(Feature(
+        #       geometry=Point([float(xx), float(-30.0)]),
+        #       properties={"name": "tmp"},
+        #   ))
+
+        for source in query_results["sources"]:
+            point = Point((source["ra"], source["dec"]))
+            aliases = [alias for alias in (source["alias"] or []) if alias]
+            source_name = ", ".join(
+                [
+                    source["id"],
+                ]
+                + aliases
+            )
+
+            features.append(
+                Feature(
+                    geometry=point,
+                    properties={
+                        "name": source_name,
+                        "url": f"/source/{source['id']}",
+                    },
+                )
+            )
+
+        query_results["geojson"] = {
+            'type': 'FeatureCollection',
+            'features': features,
+        }
+
+    return query_results
+
+
+def post_source(data, user_id, session):
+    """Post source to database.
+    data: dict
+        Source dictionary
+    user_id : int
+        SkyPortal ID of User posting the GcnEvent
+    session: sqlalchemy.Session
+        Database session for this transaction
+    """
+
+    user = session.query(User).get(user_id)
+
+    obj_already_exists = Obj.get_if_accessible_by(data["id"], user) is not None
+    schema = Obj.__schema__()
+
+    ra = data.get('ra', None)
+    dec = data.get('dec', None)
+
+    if ra is None and not obj_already_exists:
+        raise AttributeError("RA must not be null for a new Obj")
+
+    if dec is None and not obj_already_exists:
+        raise AttributeError("Dec must not be null for a new Obj")
```

Comment:
Lets combine these into one conditional and one error message...

Reply:
yes.